solutions/src/main/java/ru/sberbank/school/task02/ExtendedCalculator.java

```diff
@@ -0,0 +1,83 @@
+package ru.sberbank.school.task02;
+
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.math.RoundingMode;
+import java.util.*;
+
+import ru.sberbank.school.task02.util.*;
+
+/**
+ * Реализация обратного валютного калькулятора.
+ */
+public class ExtendedCalculator extends Calculator implements ExtendedFxConversionService {
+
+    public ExtendedCalculator(ExternalQuotesService externalQuotesService) {
+        super(externalQuotesService);
+    }
+
+    @Override
+    public Optional<BigDecimal> convertReversed(ClientOperation operation, Symbol symbol, BigDecimal amount,
+                                                Beneficiary beneficiary) {
+
+        List<Quote> quotes = externalQuotesService.getQuotes(symbol);
+        if (quotes == null || quotes.isEmpty()) {
+            throw new EmptyQuoteException("No quotes!");
+        }
+
+        List<ReverseQuote> reverseQuotes = getReverseQuotes(quotes, operation);
+
+        List<BigDecimal> prices = new ArrayList<>();
+
+        for (ReverseQuote r : reverseQuotes) {
+            if (r.getVolumeFrom().isInfinity() && amount.compareTo(r.getVolumeToSize()) < 0 ||
```

Comment:
&#39;||&#39; should be on a new line.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1313681704>*